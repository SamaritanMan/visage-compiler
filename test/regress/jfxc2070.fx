/**
 * Regression test JFXC-2070 : Compilation of script fails if String sequence contains null some times
 *
 * @test
 * @run
 */

var sseq:String[];
sseq = ["JavaFX",null];
println(sseq);
sseq = ["JavaFX", if (false) "BAD" else null];
println(sseq);

var bsseq:String[] = bind ["JavaFX",null];
println(bsseq);
var bsseq2:String[] = bind ["JavaFX", if (false) "BAD" else null];
println(bsseq2);

var oseq:Object[];
oseq = ["JavaFX",null];
println(oseq);
oseq = ["JavaFX",fo(null)];
println(oseq);

var boseq:Object[] = bind ["JavaFX", null];
println(boseq);
var boseq2:Object[] = bind ["JavaFX", fo(null)];
println(boseq2);

function fo(o:Object):Object {
  o
}
