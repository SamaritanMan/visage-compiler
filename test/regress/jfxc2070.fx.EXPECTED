[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
[ JavaFX ]
